Qt: Adds time and date + group OSD checkboxes

RedDevilus opened the pull request:
### Description of Changes
<!-- Brief description or overview on what was changed in the PR -->
Adds the options system time and system date in the OSD section of Qt and also FSUI for Big Picture mode.

Done some preliminary work on grouping all the OSD checkboxes including the new ones in a better to glance 3 column viewscreen. Also keep in mind I enabled all the OSD options as an example, no regular user will do that hence it will feel a bit cluttered at a glance with these below screenshots. V1 is the one in the code but I can change it later in future if need be.

Master:
- ![image](https://github.com/user-attachments/assets/419b1060-b434-449c-8fb4-d46554b13036)

PR:
- ![2025-05-10_20-38](https://github.com/user-attachments/assets/324a269f-e551-4453-8c40-f2cc1b6e4a8c)

V1 Chosen Minimal:

![image](https://github.com/user-attachments/assets/1d38d762-a1e0-457d-a1f4-0c9e5ffa95bd)


V2 Minimal:
- ![2025-05-10_04-43](https://github.com/user-attachments/assets/1b345c13-e4f3-433b-8ec4-cd880cf6be6f)

V3 Mix letters and numbers:
- ![V2OSDTimeDate](https://github.com/user-attachments/assets/4037aaac-d134-401e-ac56-a45f3400773d)

V4 V2 basically with format in parentheses:
- ![V3OSDTimeDate](https://github.com/user-attachments/assets/b9683afe-8941-4ec9-99fb-83c4e1d34109)





### Rationale behind Changes
<!-- Why were these changes made?  What problem does it solve / area does it improve? -->
While there is already date on the gamelist and clock in the FSUI, I find it lacks control whilst in-game. Be it for easy recording, sharing screen or maybe even speedrunning. In future it can also be tracked per locale because countries can have a mix of / or - or even dots to others to a different sorting of day, month and year. In future I can also add clock and calendar icons.

I want it more like RTSS or other overlays where you can get stuff like this:

- ![image](https://github.com/user-attachments/assets/4f55ff7c-f05c-488e-9470-0e464d1c8d44)

And you won't enable every option on there either.

Current version can be adjusted also like this:
- ![image](https://github.com/user-attachments/assets/211d1df8-3e60-444a-bab5-53a05f117e30)


I want to do dropshadows on the text in future as well, if there is white-ish background you can barely see that as well, so it's also a smaller PR for another future PR.

### Suggested Testing Steps
<!-- If applicable, including examples you've already tested with / recommendations for how to test further is very helpful! -->
Test in Big Picture Mode, global and per-game settings on multiple OSes.

RedDevilus committed:
Qt: Adds time and date + group OSD checkboxes

Adds the options system time and system date in the OSD section of Qt and also FSUI for Big Picture mode.

Done some preliminary work on grouping all the OSD checkboxes including the new ones in a better to glance 3 column viewscreen.

TheTechnician27 commented:
I'm usually opposed to changing the OSD settings, but I like headers for categories of settings. Right now, if the user doesn't just check all the boxes (maybe we should have one for 'enable all'?), they have to scan through every option until they find the one(s) they want. Likewise, system date and time are clearly relevant information for TAS and plausibly relevant for real-time players. I could probably bikeshed about the exact layout of the settings, but I think both the two new settings and the inclusion of headers are positive steps.

Bikeshedding: I wonder if there's a better name for "Show Statistics" and "Show Indicators", but I can't say for sure; maybe they're too broad to have a good, precise name, and that isn't this PR's problem anyway. I also think 'Warnings' is better than 'Warnings for user'.

RedDevilus commented:
> I'm usually opposed to changing the OSD settings, but I like headers for categories of settings. Right now, if the user doesn't just check all the boxes (maybe we should have one for 'enable all'?), they have to scan through every option until they find the one(s) they want. Likewise, system date and time are clearly relevant information for TAS and plausibly relevant for real-time players. I could probably bikeshed about the exact layout of the settings, but I think both the two new settings and the inclusion of headers are positive steps.
> 
> Bikeshedding: I wonder if there's a better name for "Show Statistics" and "Show Indicators", but I can't say for sure; maybe they're too broad to have a good, precise name, and that isn't this PR's problem anyway. I also think 'Warnings' is better than 'Warnings for user'.

Yeah, I wanted to add a global toggle for the OSD checkboxes, maybe even per column (3) but I didn't want to blast with feature creep and going with massive changes. Just wanted to test some waters and then adjust some more in the future like having no dropshadows on the text annoys me that I can barely see stuff, maybe even make it so transparency can be adjusted or even the colors but that's just some ideas I might want to incorporate.

F0bes commented:
I think having the date and time is a silly addition.

RedDevilus commented:
Nobody is going to press Escape constantly to check the time or alt tab if it's fullscreen. Never said it was essential ,but I personally think it has a place and if you don't need it you can also not enable it.

Mrlinkwii commented:
i have a feeling the OSD is becoming more bloated with this  , if someone really needs the time they can run RTSS or similar tools

RedDevilus committed:
Qt: Fix typo and place them in same line + remove Time: and Date: notation

Currently typo fixed and instead of Time and Date being on seperate lines and then speedline, use a pipe for divider for same line. Also got rid of Time: and Date: being displayed it is kinda redudant.

JordanTheToaster reviewed (approved):
I can now see the time and date while being overstimulated by OSD information.

TheTechnician27 commented on pcsx2-qt/Settings/GraphicsSettingsWidget.cpp:
```diff
@@ -799,6 +802,15 @@ GraphicsSettingsWidget::GraphicsSettingsWidget(SettingsWindow* dialog, QWidget*
 			tr("Shows the current system hardware information on the OSD."));
 
+		dialog->registerWidgetHelp(m_ui.osdShowSystemTime, tr("Show System Time"), tr("Unchecked"),
+			tr("Shows the current system time in the top-right corner of the display currently with Hour:Minute:Second (HH:MM:SS) format."));
```
Should be "Shows the current system time in Hour:Minute:Second (HH:MM:SS) format." Top-right is no longer static, and "current" is repeated ("current system time" and "display currently") Same with the next two dialogs. This static "top-right" needs to be fixed elsewhere too.

kamfretoz reviewed (commented):
Just a small nitpick

RedDevilus committed:
Qt: Capitalise Warnings For User

RedDevilus commented:
> ![RTC](https://private-user-images.githubusercontent.com/12673964/462940259-4a40f47c-d827-4d2a-a87c-403d646356d9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTE4MTk3NjIsIm5iZiI6MTc1MTgxOTQ2MiwicGF0aCI6Ii8xMjY3Mzk2NC80NjI5NDAyNTktNGE0MGY0N2MtZDgyNy00ZDJhLWE4N2MtNDAzZDY0NjM1NmQ5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzA2VDE2MzEwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE1MjI0ZjU5MDQwZmFkZDcxYzZkZDIwNmIxNmYxMmJiOTQ0MmIwYWIyZjQwMWUxNWZhMzFiM2M1MjNiNWY3MzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0._CtRtrqrk-RMHTzRmx2MrF9jwfr5uHDHThgJFvoXrU8)
> 
> ![System Time](https://private-user-images.githubusercontent.com/12673964/462940314-f2ce02c0-f4b7-4b99-b3f1-d4d8c83fe7c3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTE4MTk3NjIsIm5iZiI6MTc1MTgxOTQ2MiwicGF0aCI6Ii8xMjY3Mzk2NC80NjI5NDAzMTQtZjJjZTAyYzAtZjRiNy00Yjk5LWIzZjEtZDRkOGM4M2ZlN2MzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzA2VDE2MzEwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQzNDdjNjk3ZjdkMzVlNThhOTMxNTRiMjY2ZTBiNjIwNWI3MjlhMjI4ZTdiMDZjNjNiZjkzZGI5ZTVjYjQ4NTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.igV-Elazsp0TfKKb6fUQrKb2aAcc5TL6fsRfVsrJquE)
> 
> It looks like the PR only fetches system time but does not account for the possibility the user is using an RTC other than the default system time one.

Not sure if it should follow RTC perse as it doesn't follow the PS2 internal clock and date (desync if you press Tab or Press F4). This PR syncs up with system time and yes on boot in BIOS in PCSX2 it syncs up everything once. Maybe a better method exists in future PR where we can have PS2 System Clock vs PC System Clock being obvious.

TheTechnician27 commented:
I disagreed with fobes' comment initially because I thought this was for the RTC time and date, not the desktop time and date. Even though the PR works well for what it intends to do and needs just a few tweaks functionality-wise, that information is trivial for a user to access outside of PCSX2 and unfortunately feels very "I want PCSX2 to file my taxes".
